tests/test_iqm_provider.py

```diff
@@ -121,10 +129,20 @@ def test_qubit_name_to_index_to_qubit_name(adonis_architecture_shuffled_names):
     assert backend.qubit_name_to_index('Alice') is None
 
 
-def test_serialize_circuit_raises_error_for_non_transpiled_circuit(backend, circuit):
-    circuit = QuantumCircuit(2, 2)
-    with pytest.raises(ValueError, match='has not been transpiled against the current backend'):
-        backend.serialize_circuit(circuit)
+def test_serialize_circuit_raises_error_for_non_transpiled_circuit(circuit, linear_architecture_3q):
+    client = IQMClient(url='http://some_url')
+    client._token_manager = None  # Do not use authentication
+    when(client).get_quantum_architecture().thenReturn(linear_architecture_3q)
+    expect(requests, times=1).post(client._base_url, ANY).thenReturn(
+        MockJsonResponse(201, {'id': str(uuid.UUID('3c3fcda3-e860-46bf-92a4-bcc59fa76ce9'))})
+    )
```

Comment:
The `expect(requests, times=1)` does not actually check that it's called 1 time unless there is also `verifyNoUnwantedInteractions()` at the end of the test (https://mockito-python.readthedocs.io/en/latest/the-functions.html#mockito.expect). If that is added, you can see that this request is in fact never sent during this test, because the error is raised before that. I would suggest just removing this part (and MockJsonResponse class since it's not used elsewhere).

```suggestion
```